### PR TITLE
Add protected logger to SpringBootCondition

### PR DESCRIPTION
**Pull request by olegz**

### CODE REVIEW

File: spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/condition/SpringBootCondition.java [hunk 1]
```diff
@@ -37,7 +37,7 @@
  */
 public abstract class SpringBootCondition implements Condition {
 
-	private final Log logger = LogFactory.getLog(getClass());
+	protected final Log logger = LogFactory.getLog(getClass());
```
Comment (snicoll) [inline spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/condition/SpringBootCondition.java]: I think I'd be more in favour of a `protected getLogger()` method.

Comment (olegz) [inline spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/condition/SpringBootCondition.java]: @snicoll FWIW, I thought of that and did check all references I could find, but couldn't find any at least in spring-boot space.
As for the `getLogger()`, I am ok with that, will update pr

**Comment by philwebb** [comment]
I'm not too keen on making the logger part of the core API. Spring Framework has a few of these and it means that `commons-logging` can never be replaced without breaking binary compatibility.

**Comment by snicoll** [comment]
@philwebb alright. I've revert the original change.

**Pull request closed without merging**